RELEASE.md

```diff
@@ -4,90 +4,81 @@
-
-## Prep the release (part 1)
+## Prep the release
 
 1. Ensure the Travis build is passing against master.
 
-2. Update the version strings (1.2.3-dev) to the new version (1.2.4). There should only be one place this needs to be done: ``skbio/__init__.py``. It's a good idea to ``grep`` for the current version string just to be safe:
+2. Update the version strings (1.2.3-dev) to the new version (1.2.4). This will include `__version__` defined in ``skbio/__init__.py``, as well as any `@experimental/@stable/@deprecated` [API stability decorators](http://scikit-bio.org/docs/latest/user/api_stability.html) with `as_of='1.2.3-dev'`. ``grep`` for the current version string to find all occurrences:
+
+        grep -r '1\.2\.3-dev' .
+
+3. Remove any deprecated functionality that was scheduled for removal on or before this release. When removing deprecated functionality, make sure the functionality has been in a deprecated state for the appropriate number of releases described in the [API stability docs](http://scikit-bio.org/docs/latest/user/api_stability.html). If there is functionality that shouldn't be removed yet, bump the `until` version to a future version. To find all deprecated functionality, search for `@deprecated` decorators:
 
-        grep -ir '1\.2\.3-dev' *
+        grep -r '@deprecated' .
 
-3. Update ``CHANGELOG.md`` to include descriptions of the changes that made it into this release. Be sure to update the heading to include the new version (1.2.4) and the date of the release. Use the existing structure in the file as a template/guide.
+    Note any deprecated functionality that was removed in the `### Miscellaneous` section of `CHANGELOG.md`.
 
-4. Submit a pull request with these changes and let Travis run.
+4. Update ``CHANGELOG.md`` to include descriptions of all **user-facing** changes that made it into this release. Be sure to update the heading to include the new version (1.2.4) and the date of the release. Use the existing structure in the file as a template/guide.
 
-## Build the documentation
+5. Submit a pull request and merge when Travis-CI tests are passing.
 
-In the meantime, you can build the documentation and update the website.
+## Build website docs
 
-**Note:** You will need to **fully install** (including built extensions) the exact version of scikit-bio that you are editing so that Sphinx will pull docstrings from the correct version of the code. **Make sure the version of scikit-bio that is imported by ``import skbio`` is the correct one!**
+You will need to **fully install** the latest master branch of scikit-bio (including built extensions) and build the docs from this version. Make sure the version of scikit-bio that is imported by ``import skbio`` is the correct one before building the docs.
```

Comment:
I would bold the last sentence, if only because I have messed this part up countless times myself.